engine/src/main/java/org/terasology/rendering/nui/widgets/UICheckbox.java

```diff
@@ -15,39 +15,30 @@
  */
 package org.terasology.rendering.nui.widgets;
 
-import com.google.common.collect.Lists;
 import org.terasology.input.MouseInput;
 import org.terasology.math.geom.Vector2i;
+import org.terasology.rendering.nui.ActivateableWidget;
 import org.terasology.rendering.nui.BaseInteractionListener;
 import org.terasology.rendering.nui.Canvas;
-import org.terasology.rendering.nui.CoreWidget;
 import org.terasology.rendering.nui.InteractionListener;
 import org.terasology.rendering.nui.databinding.Binding;
 import org.terasology.rendering.nui.databinding.DefaultBinding;
 import org.terasology.rendering.nui.events.NUIMouseClickEvent;
 
-import java.util.List;
-
 /**
  * A check-box. Hovering is supported.
  */
-public class UICheckbox extends CoreWidget {
+public class UICheckbox extends ActivateableWidget {
     public static final String HOVER_ACTIVE_MODE = "hover-active";
 
     private Binding<Boolean> active = new DefaultBinding<>(false);
 
-    /**
-     * A {@link List} of listeners subscribed to this checkbox
-     */
-    private List<ActivateEventListener> listeners = Lists.newArrayList();
-
     private InteractionListener interactionListener = new BaseInteractionListener() {
 
         @Override
         public boolean onMouseClick(NUIMouseClickEvent event) {
             if (event.getMouseButton() == MouseInput.MOUSE_LEFT) {
                 active.set(!active.get());
-                activate();
```

Comment:
You, uh, broke checkboxes with these changes.
Most likely you forgot to call the new `activateWidget` method. You also override it below so you'll need to call super there